kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionSidePanel.vue

```diff
@@ -27,69 +34,63 @@
     components: {
       SidePanelModal,
     },
-    setup(_, context) {
+    setup() {
       const store = getCurrentInstance().proxy.$store;
       const router = getCurrentInstance().proxy.$router;
       const route = computed(() => store.state.route);
 
-      const routeWhenSidePanelOpened = route.value;
-      const prevRoute = ref({
-        name: PageNames.EXAM_CREATION_ROOT,
-        params: { ...route.value.params },
-      });
-      const canCloseSidePanel = ref(true);
+      const canGoBack = ref(false);
       const showSidePanel = computed(() => route.value?.name !== PageNames.EXAM_CREATION_ROOT);
 
-      watch(route, (to, from) => {
-        // We're on the same route we were on when we started so should be able to close it
-        canCloseSidePanel.value =
-          to.name === routeWhenSidePanelOpened.name &&
-          // I tried using _.isEqual on the params & query objects but it didn't work as I expected
-          // so this is hard-coded for now.
-          to.params.sectionIndex === routeWhenSidePanelOpened.params.sectionIndex &&
-          to.params.topic_id === routeWhenSidePanelOpened.params.topic_id &&
-          to.query.showBookmarks === routeWhenSidePanelOpened.query.showBookmarks &&
-          to.query.search === routeWhenSidePanelOpened.query.search;
-
-        prevRoute.value = from;
-      });
-
       function handleClosePanel() {
-        if (canCloseSidePanel.value) {
-          // Avoid redundant navigation error
-          if (prevRoute.value.name === PageNames.EXAM_CREATION_ROOT) {
-            router.back();
-          } else {
-            router.push({
-              name: PageNames.EXAM_CREATION_ROOT,
-              params: { ...prevRoute.value.params },
-            });
-          }
-        } else {
-          router.back();
-        }
+        router.push({
+          name: PageNames.EXAM_CREATION_ROOT,
+          params: { ...route.value.params },
```

Comment:
Should be specific here, see e.g. https://github.com/learningequality/kolibri/blob/8d285cfebff3c91d9a2e1ee273d1dfc8dab72184/kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/ReplaceQuestions.vue#L241

(although I noticed in searching for this, there are still lots of places just doing a general spread).

Because of the way Vue router works, items that are not in the URL still end up in the params state, which can cause unexpected bugs down the line.